### ISSUE TITLE
data.gouv.fr data import needs to understand devices with multiple connectors

### ISSUE THREAD

**Issue by zymurgic** [openchargemap/ocm-system#160]
Noticed that the dataset has in many cases a row per connector, so for instance, the device with the prefix
FR*V07*E72264*003
has three connectors:
FR*V07*E72264*003*1 : Type 2 AC
FR*V07*E72264*003*2 : Chademo
FR*V07*E72264*003*3 : CCS Type 2
note: Those rows are not consequetive in the import data.
Currently, due to the proximity of existing stations in the import code, only the first of those encountered in the file got imported.
We probably need to use the first four asterisk-delimited fields of the station identifier to group them to the logical device, and the full station identifier becomes the per-connector identifier.

**Comment by zymurgic** [comment]
oops. looks like the github markdown messed up the asterisk-delimited fields there:

noticed that the dataset has in many cases a row per connector, so for instance, the device with the prefix
`FR*V07*E72264*003`
has three connectors:
```
FR*V07*E72264*003*1 : Type 2 AC
FR*V07*E72264*003*2 : Chademo
FR*V07*E72264*003*3 : CCS Type 2
```
note: Those rows are not consequetive in the import data.
Currently, due to the proximity of existing stations in the import code, only the first of those encountered in the file got imported.

**Comment by zymurgic** [comment]
This might possibly be a useful feature on any import that handles OCHP/OCPP identifiers, not just data.gouv.fr

**Comment by webprofusion-chrisc** [comment]
Thanks, yes we could compile the list then do a merge pass to consolidate rather than discard them as duplicates - I seem to remember one of the other imports does just that but I can't remember which (maybe UK NCR?).

**Comment by webprofusion-chrisc** [comment]
Note also that the source data does specify multiple connectors in some rows, it's a bit of a jumble and some of it is just broken data.

**Comment by zymurgic** [comment]
I've seen that for many installs in France, alongside DC rapid chargers, which have the usual access control and networked complexity, there's often an uncontrolled domestic socket on the rear of the unit for emergency charging in case the network or DC power electronics is faulty. How those show up in that dataset is unknown.

I see what you mean by a jumble of complete ,and incomplete data, for instance, if you look for 
-2.045631  ; 48.474826
T2 Combo Chademo
then that one row implies a triple-headed Chademo/CCS type 2/Type 2 at 50kW
only Type 2 only goes up to 43kW
and If it's over 22kW then it's got to be tethered.
but in some cases these triple-headed chargers only have a 22kW AC socket, so it's unclear, but we can probably guess it's got a 43kW AC Tethered type 2, 3 phase.

Those listed as 
T2-E/F
, eg, `FR*S56*P56121*P` Place d'Armes 56100 Lorient will have both a Type 2 and a domestic socket, compatible with type E or F plugs.
possibly multiple of them. Either way you can't expect more then 3.6kW out a type E or F domestic socket, and indeed very few of them set up for EV charging purposes will have a lower available power than 3kW.
and in many cases, there's no way of telling from that data what the capacity of the Type 2 socket is (looking at other, not necessarily open data for that example, I can tell that they're 32A/22kW three phase AC type 2 sockets, and and two domestic sockets at 16A/3.6kW on a single logical device).

In summary, the dataset shows signs of coming from multiple sources, and each data provider has their own quirks as to how they represent the data.

**Comment by webprofusion-chrisc** [comment]
Closing - no plans to repeat this import.
